include/mbgl/renderer/renderer.hpp

```diff
@@ -49,6 +55,20 @@ class Renderer {
                                                  const std::string& extensionField,
                                                  const optional<std::map<std::string, Value>>& args = {}) const;
 
+    void setFeatureState(const std::string& sourceID,
+                         const optional<std::string>& sourceLayerID,
+                         const std::string& featureID,
+                         const style::conversion::Convertible& state);
```

Comment:
Asheem made a good point about convertible, we will need to do an extra step to convert it to something usable.
NSDictionary -> convertible -> unordered_map